### PR TITLE
=str Add missing abstract modifier to ActorPublisherMessage.Cancel

### PR DESCRIPTION
**Pull request by RichardBradley**
This marker base class is missing an "abstract" modifier.

This is causing "warning: match may not be exhaustive" for me in certain cases

### CODE REVIEW

**Comment by drewhk** [comment]
OK TO TEST

File: akka-stream/src/main/scala/akka/stream/actor/ActorPublisher.scala [hunk 1]
```diff
@@ -52,7 +52,7 @@ object ActorPublisherMessage {
    * subscription.
    */
   final case object Cancel extends Cancel with NoSerializationVerificationNeeded
-  sealed class Cancel extends ActorPublisherMessage
+  sealed abstract class Cancel extends ActorPublisherMessage
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/actor/ActorPublisher.scala]: it looks like we are missing the Java API for `SubscriptionTimeoutExceeded`. Would you mind adding that also?
It should be similar to the below `cancelInstance`.
Please also move those close together with classes/objects.

Comment (RichardBradley) [inline akka-stream/src/main/scala/akka/stream/actor/ActorPublisher.scala]: OK, have done.

**Comment by patriknw** [comment]
LGTM, thanks

**Comment by drewhk** [comment]
Thanks, merging.

**Pull request merged**